pylutron/__init__.py

```diff
@@ -162,10 +178,13 @@ def _main_loop(self):
           line = t.read_until(b"\n", timeout=3)
         else:
           raise EOFError('Telnet object already torn down')
-      except (EOFError, TimeoutError, socket.timeout, AttributeError):
+      except _EXPECTED_NETWORK_EXCEPTIONS:
+        _LOGGER.exception("Uncaught exception")
         try:
           self._lock.acquire()
```

Comment:
aside any reason a context manager isnt used here? i haven't written threaded code in a while but looks like it'd be safe